src/drivers/imu/mpu9250/mpu9250.cpp

```diff
@@ -457,24 +484,48 @@ int MPU9250::reset()
 
 int MPU9250::reset_mpu()
 {
-	write_reg(MPUREG_PWR_MGMT_1, BIT_H_RESET);
-	write_checked_reg(MPUREG_PWR_MGMT_1, MPU_CLK_SEL_AUTO);
-	write_checked_reg(MPUREG_PWR_MGMT_2, 0);
-
+    uint8_t retries;
+
+	switch(_device_type) {
+	case MPU_DEVICE_TYPE_MPU9250:
+		write_reg(MPUREG_PWR_MGMT_1, BIT_H_RESET);
+		write_checked_reg(MPUREG_PWR_MGMT_1, MPU_CLK_SEL_AUTO);
+		write_checked_reg(MPUREG_PWR_MGMT_2, 0);
+		break;
+	case MPU_DEVICE_TYPE_ICM20948:
+		write_reg(ICMREG_20948_PWR_MGMT_1, BIT_H_RESET);
+		/*
+		 * ICM20948 needs a bit of time here, else register settings will be garbled.
+		 */
+	    usleep(200);
 
-	usleep(1000);
```

Comment:
@flochir I tried putting this delay back and at least I can get the mpu9250 to run now. Obviously to test it I uncommented the gyro integrator and uorb publishing that you commented out for debugging in the measure function.

Reply:
if this is necessary as it seems, we can reschedule the reset_mpu using the work queue

Reply:
Thanks! Can you try again with my second latest commit (8481016873e4b640801a8502c9de9807b41c0a4c, the latest doesn't publish sensor data)? The mpu9250 is working for me with this one, but with only the two short delays as up_udelay().
Rescheduling probably still makes sense later on, this could be called after setup if the sensor or the bus have problems and a reset is triggered.